sci-astronomy/siril/siril-1.0.2.ebuild

```diff
@@ -0,0 +1,70 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit meson toolchain-funcs xdg
+
+DESCRIPTION="A free astronomical image processing software"
+HOMEPAGE="https://www.siril.org/"
+
+if [[ ${PV} == "9999" ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://gitlab.com/free-astro/${PN}.git"
+else
+	SRC_URI="https://gitlab.com/free-astro/siril/-/archive/${PV/_/-}/${PN}-${PV/_/-}.tar.bz2"
+	KEYWORDS="~amd64 ~x86"
+	S="${WORKDIR}/${PN}-${PV/_/-}"
+fi
+
+LICENSE="GPL-3+"
+SLOT="0"
+IUSE="curl ffmpeg gnuplot heif jpeg openmp png raw tiff wcs"
+
+DEPEND="
+	>=dev-libs/glib-2.56.0:2
+	>=dev-libs/json-glib-1.2.6
+	>=dev-libs/libconfig-1.4[cxx]
+	>=media-gfx/exiv2-0.25
+	media-libs/librtprocess:=
+	>=media-libs/opencv-4.4.0:=
+	sci-libs/cfitsio
+	sci-libs/fftw:3.0=
+	sci-libs/gsl:=
+	x11-libs/cairo
+	>=x11-libs/gtk+-3.20.0:3
+	curl? ( net-misc/curl )
+	ffmpeg? ( media-video/ffmpeg:= )
+	heif? ( media-libs/libheif )
+	jpeg? ( media-libs/libjpeg-turbo:= )
+	png? ( >=media-libs/libpng-1.6.0 )
+	raw? ( media-libs/libraw )
+	tiff? ( media-libs/tiff )
+	wcs? ( >=sci-astronomy/wcslib-7.7 )
```

Comment:
These don't have any function in this ebuild. You should remove them as use flags and add as `optfeature` instead.
https://projects.gentoo.org/qa/policy-guide/dependencies.html#pg0001

Reply:
Unfortunately the function of this USE flags is not obvious, but they actually influence the build process. All the above mentioned dependencies are optional (refer to https://gitlab.com/free-astro/siril#requirements). But I didn't figured out a way to switch the appropriate options in the meson build system.

Instead meson goes the other way round. It checks whether an optional dependency is installed at build time and then enabled a feature. So to compile in WCS support for example, `sci-astronomy/wcslib` has to be present at build time. Just installing it later wouldn't work.

Does your remark still apply under this circumstances? I don't think so. But maybe I am wrong and in this case please correct me.

Reply:
Unfortunately that needs to be patched to accept configure options then - since this kind of "automagic" behaviour is dangerous:
https://wiki.gentoo.org/wiki/Project:Quality_Assurance/Automagic_dependencies
https://wiki.gentoo.org/wiki/Project:Quality_Assurance/Automagic_dependencies#Why_automagic_dependencies_are_wrong

We should update the wiki page to have an example about meson. But look at how libcurl is enabled in meson.build and try to apply that to the other optional deps? And yes, this is something that should be fixed upstream.

Reply:
Thanks for the advice. I will try to fix it, but it will take some time. In the meantime I will switch this PR to the draft state.

Reply:
I prepared a patch set. I will try to get this patch upstream.